include/RAJA/pattern/params/forall.hpp

```diff
@@ -43,45 +43,45 @@ struct ForallParamPack
 private:
   // Init
   template<typename EXEC_POL, camp::idx_t... Seq, typename... Args>
-  static constexpr void detail_init(EXEC_POL,
+  static constexpr void params_init(EXEC_POL const& pol,
                                     camp::idx_seq<Seq...>,
                                     ForallParamPack& f_params,
                                     Args&&... args)
   {
-    CAMP_EXPAND(expt::detail::init<EXEC_POL>(camp::get<Seq>(f_params.param_tup),
-                                             std::forward<Args>(args)...));
+    CAMP_EXPAND(param_init(pol, camp::get<Seq>(f_params.param_tup),
+                           std::forward<Args>(args)...));
   }
 
   // Combine
   template<typename EXEC_POL, camp::idx_t... Seq>
-  RAJA_HOST_DEVICE static constexpr void detail_combine(
-      EXEC_POL,
+  RAJA_HOST_DEVICE static constexpr void params_combine(
+      EXEC_POL const& pol,
       camp::idx_seq<Seq...>,
       ForallParamPack& out,
       const ForallParamPack& in)
   {
-    CAMP_EXPAND(detail::combine<EXEC_POL>(camp::get<Seq>(out.param_tup),
-                                          camp::get<Seq>(in.param_tup)));
+    CAMP_EXPAND(param_combine(pol, camp::get<Seq>(out.param_tup),
+                              camp::get<Seq>(in.param_tup)));
   }
 
   template<typename EXEC_POL, camp::idx_t... Seq>
-  RAJA_HOST_DEVICE static constexpr void detail_combine(
-      EXEC_POL,
+  RAJA_HOST_DEVICE static constexpr void params_combine(
+      EXEC_POL const& pol,
       camp::idx_seq<Seq...>,
       ForallParamPack& f_params)
   {
-    CAMP_EXPAND(detail::combine<EXEC_POL>(camp::get<Seq>(f_params.param_tup)));
+    CAMP_EXPAND(param_combine(pol, camp::get<Seq>(f_params.param_tup)));
```

Comment:
Is the difference between `params_combine` and `param_combine` just that the former is at the loop level, and the latter is in the backend? This naming can be a little confusing and possibly error prone if we mistype a recursive call. Maybe something like `local_params_combine` or `params_combine_impl` for the backend one?

Reply:
Ya, one calls combine on a parameter pack and one is the implementation for a single parameter. More clear differentiation is probably a good idea. How about something like `parampack_combine`?

Reply:
`parampack_comine` and `param_combine`? That should be much more clear.

Reply:
done